asyncpg/transaction.py

```diff
@@ -111,19 +112,29 @@ async def start(self):
         else:
             # Nested transaction block
             top_xact = con._top_xact
-            if self._isolation != top_xact._isolation:
-                raise apg_errors.InterfaceError(
-                    'nested transaction has a different isolation level: '
-                    'current {!r} != outer {!r}'.format(
-                        self._isolation, top_xact._isolation))
+            if top_xact._isolation is None:
+                if self._isolation:
```

Comment:
Perhaps run `SHOW transaction_isolation;` here to determine the outer transaction isolation level?

Reply:
Yeah, that'll work, I was only uncertain about whether we want an extra query - but I think you're probably right that this is already a corner case where isolation is specified for a nested transaction while the outer transaction's isolation is unspecified. Let me add that in a new commit.